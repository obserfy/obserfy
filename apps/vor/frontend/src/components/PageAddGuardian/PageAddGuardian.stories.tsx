import { FC } from "react"
import PageAddGuardian from "./PageAddGuardian"

export default {
  title: "Pages/PageAddGuardian",
  component: PageAddGuardian,
  parameters: {
    componentSubtitle: "Just a simple Add guardian page",
  },
}

export const Basic: FC = () => <PageAddGuardian id="asdfsdaf" />
