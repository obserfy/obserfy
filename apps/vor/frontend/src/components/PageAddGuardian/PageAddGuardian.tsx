import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Flex } from "theme-ui"
import { getFirstName } from "../../domain/person"
import { usePostGuardianRelation } from "../../hooks/api/guardians/usePostGuardianRelation"
import { usePostNewGuardian } from "../../hooks/api/guardians/usePostNewGuardian"
import { GuardianRelationship } from "../../hooks/api/students/usePostNewStudent"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import {
  STUDENT_OVERVIEW_URL,
  STUDENT_PROFILE_URL,
  STUDENTS_URL,
} from "../../routes"
import Chip from "../Chip/Chip"
import GuardianRelationshipSelector from "../GuardianRelationshipSelector/GuardianRelationshipSelector"
import { navigate } from "../Link/Link"
import NewGuardianForm, {
  useNewGuardianFormState,
} from "../NewGuardianForm/NewGuardianForm"
import SimpleGuardiansSelector from "../SimpleGuardiansSelector/SimpleGuardiansSelector"
import { breadCrumb } from "../TopBar/TopBar"
import TopBarWithAction from "../TopBarWithAction/TopBarWithAction"
import { Typography } from "../Typography/Typography"

enum Mode {
  NEW,
  EXISTING,
}

interface Props {
  id: string
}

export const PageAddGuardian: FC<Props> = ({ id: studentId }) => {
  const { data: student } = useGetStudent(studentId)
  const postNewGuardian = usePostNewGuardian(studentId)

  const [mode, setMode] = useState(Mode.NEW)
  const [relation, setRelation] = useState(GuardianRelationship.Mother)
  const [newGuardian, setNewGuardian] = useNewGuardianFormState()
  const [guardianId, setGuardianId] = useState("")
  const postNewGuardianRelation = usePostGuardianRelation(
    { id: guardianId },
    studentId
  )

  const isFormValid =
    mode === Mode.NEW ? newGuardian.name !== "" : guardianId !== ""

  const createNewGuardian = async () => {
    try {
      await postNewGuardian.mutateAsync({
        ...newGuardian,
        studentId,
        relationship: relation,
      })
      await navigate(STUDENT_PROFILE_URL(studentId))
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  const createNewGuardianRelation = async () => {
    try {
      await postNewGuardianRelation.mutateAsync(relation)
      await navigate(STUDENT_PROFILE_URL(studentId))
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <>
      <TopBarWithAction
        disableAction={!isFormValid}
        onActionClick={async () => {
          if (mode === Mode.NEW) await createNewGuardian()
          else await createNewGuardianRelation()
        }}
        breadcrumbs={[
          breadCrumb(t`Students`, STUDENTS_URL),
          breadCrumb(getFirstName(student), STUDENT_OVERVIEW_URL(studentId)),
          breadCrumb(t`Profile`, STUDENT_PROFILE_URL(studentId)),
          breadCrumb(t`Add Guardian`),
        ]}
        isLoading={postNewGuardian.isLoading}
      />

      <Box mx="auto" sx={{ maxWidth: "maxWidth.sm" }}>
        <Typography.H5 mx={3} mt={4} mb={2}>
          <Trans>Add Guardian</Trans>
        </Typography.H5>

        <ModeSelector mode={mode} onClick={setMode} />

        <GuardianRelationshipSelector
          value={relation}
          onChange={(e) => setRelation(parseInt(e.target.value, 10))}
        />

        {mode === Mode.NEW && (
          <NewGuardianForm
            newGuardian={newGuardian}
            onChange={setNewGuardian}
          />
        )}
        {mode === Mode.EXISTING && (
          <SimpleGuardiansSelector
            onChange={setGuardianId}
            selectedId={guardianId}
            currentGuardianIds={student?.guardians.map(({ id }) => id)}
          />
        )}
      </Box>
    </>
  )
}

const ModeSelector: FC<{
  mode: Mode
  onClick: (mode: Mode) => void
}> = ({ mode, onClick }) => (
  <Flex mb={3}>
    <Chip
      text="Create new"
      ml={3}
      isActive={mode === Mode.NEW}
      onClick={() => onClick(Mode.NEW)}
    />
    <Chip
      text="From existing"
      ml={2}
      isActive={mode === Mode.EXISTING}
      onClick={() => onClick(Mode.EXISTING)}
    />
  </Flex>
)

export default PageAddGuardian
