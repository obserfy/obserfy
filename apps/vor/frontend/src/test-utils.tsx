/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
