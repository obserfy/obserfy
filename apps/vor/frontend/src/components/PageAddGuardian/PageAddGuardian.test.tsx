import { render } from "../../test-utils"
import { Basic } from "./PageAddGuardian.stories"

describe("PageNewGuardian", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
