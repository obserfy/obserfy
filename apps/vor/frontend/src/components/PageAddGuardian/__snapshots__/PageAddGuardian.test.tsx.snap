// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNewGuardian should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surfaceBlurNonTransparent);
  z-index: 100;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border);
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 640px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-5:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-8 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-8:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-8:last-child>.splitter {
  display: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-17:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-17:last-child>.splitter {
  display: none;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
  margin-right: 16px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-18:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-18:hover,
.emotion-18:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-18:hover:disabled,
.emotion-18:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 32px;
  margin-bottom: 8px;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: var(--theme-ui-colors-primary);
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border-radius: 6px;
  border-color: var(--theme-ui-colors-border);
  border-width: 1px;
  border-style: solid;
  white-space: nowrap;
}

.emotion-22:hover {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-onPrimary);
  font-size: 14px;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 8px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: var(--theme-ui-colors-surface);
  display: inline-block;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border-radius: 6px;
  border-color: var(--theme-ui-colors-border);
  border-width: 1px;
  border-style: solid;
  white-space: nowrap;
}

.emotion-24:hover {
  background-color: var(--theme-ui-colors-primaryLightest);
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-28 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-30 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: var(--theme-ui-colors-background);
  border-radius: 6px;
  color: var(--theme-ui-colors-text);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
}

.emotion-30:hover,
.emotion-30:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-31 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

.emotion-32 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-33 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-35 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-36 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-36:autofill,
.emotion-36:autofill:hover,
.emotion-36:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-36:autofill:first-line,
.emotion-36:autofill:hover:first-line,
.emotion-36:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-36:-webkit-autofill,
.emotion-36:-webkit-autofill:hover,
.emotion-36:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-36:-webkit-autofill:first-line,
.emotion-36:-webkit-autofill:hover:first-line,
.emotion-36:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-36:hover,
.emotion-36:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-36::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-36::-moz-placeholder {
  font-style: italic;
}

.emotion-36:-ms-input-placeholder {
  font-style: italic;
}

.emotion-36::placeholder {
  font-style: italic;
}

.emotion-36:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-44 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-44:autofill,
.emotion-44:autofill:hover,
.emotion-44:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-44:autofill:first-line,
.emotion-44:autofill:hover:first-line,
.emotion-44:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-44:-webkit-autofill,
.emotion-44:-webkit-autofill:hover,
.emotion-44:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-44:-webkit-autofill:first-line,
.emotion-44:-webkit-autofill:hover:first-line,
.emotion-44:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-44:hover,
.emotion-44:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-44::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-44::-moz-placeholder {
  font-style: italic;
}

.emotion-44:-ms-input-placeholder {
  font-style: italic;
}

.emotion-44::placeholder {
  font-style: italic;
}

.emotion-44:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-49 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-51 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 16px;
  height: 180px;
  resize: vertical;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
}

.emotion-51:hover,
.emotion-51:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-51::-webkit-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-51::-moz-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-51:-ms-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-51::placeholder {
  font-style: italic;
  line-height: 2rem;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <a
            class="emotion-4"
            data-cy="back-button"
            href="/dashboard/students/profile?studentId=asdfsdaf"
          >
            <button
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
            </button>
          </a>
          <div
            class="emotion-7"
          >
            <a
              class="emotion-8"
              href="/dashboard/students"
            >
              <span
                class="emotion-9"
              >
                Students
              </span>
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <a
              class="emotion-8"
              href="/dashboard/students/overview?studentId=asdfsdaf"
            >
              <span
                class="emotion-9"
              />
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <a
              class="emotion-8"
              href="/dashboard/students/profile?studentId=asdfsdaf"
            >
              <span
                class="emotion-9"
              >
                Profile
              </span>
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <div
              class="emotion-17"
            >
              Add Guardian
              <span
                class="splitter"
              >
                /
              </span>
            </div>
          </div>
        </div>
        <button
          class="emotion-18"
          data-cy="finish-cta"
          disabled=""
        >
          Add
        </button>
      </div>
    </div>
    <div
      class="emotion-19"
    >
      <h5
        class="emotion-20"
      >
        Add Guardian
      </h5>
      <div
        class="emotion-21"
      >
        <div
          class="emotion-22"
        >
          <p
            class="emotion-23"
          >
            Create new
          </p>
        </div>
        <div
          class="emotion-24"
        >
          <p
            class="emotion-25"
          >
            From existing
          </p>
        </div>
      </div>
      <div
        class="emotion-26"
      >
        <label
          class="emotion-27"
        >
          <div
            class="emotion-28"
          >
            Relationship
          </div>
          <div
            class="emotion-29"
          >
            <select
              aria-label="Relationship"
              class="emotion-30"
            >
              <option
                value="0"
              >
                Other
              </option>
              <option
                value="1"
              >
                Mother
              </option>
              <option
                value="2"
              >
                Father
              </option>
            </select>
            <svg
              class="emotion-31"
              fill="currentcolor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </label>
      </div>
      <div
        class="emotion-32"
      >
        <label
          class="emotion-33"
        >
          <div
            class="emotion-28"
          >
            Guardian Name
          </div>
          <div
            class="emotion-35"
          >
            <input
              class="emotion-36"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-33"
        >
          <div
            class="emotion-28"
          >
            Email
          </div>
          <div
            class="emotion-35"
          >
            <input
              class="emotion-36"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-33"
        >
          <div
            class="emotion-28"
          >
            Phone
          </div>
          <div
            class="emotion-35"
          >
            <input
              class="emotion-44"
              type="phone"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-33"
        >
          <div
            class="emotion-28"
          >
            Address
          </div>
          <div
            class="emotion-35"
          >
            <input
              class="emotion-44"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-49"
        >
          <div
            class="emotion-28"
          >
            Note
          </div>
          <textarea
            aria-label="Note"
            class="emotion-51"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
